### PR TITLE
[docs] Add additional note for license key installation on Next.js

### PR DESCRIPTION
**Pull request by joserodolfofreitas**
## Summary

Add instruction to add the `NEXT_PUBLIC` prefix to environment variables.

The pain point was previously seen at:
https://mui.zendesk.com/agent/tickets/6989

## Preview
image

https://deploy-preview-9575--material-ui-x.netlify.app/x/introduction/licensing/

### CODE REVIEW

**Review by richbustos — APPROVED** [review]
LGTM!

File: docs/data/introduction/licensing/licensing.md [hunk 1]
```diff
@@ -219,6 +219,15 @@ export default function MyApp(props) {
 }
 ```
 
+#### Install the key as an environment variable in Next.js
```
Comment (oliviertassinari) [inline docs/data/introduction/licensing/licensing.md:222]: As I understand it, the values in the .env shouldn't be in git, developers on the team have to find the value, somehow.

Screenshot 2023-07-11 at 00 04 21

https://nextjs.org/docs/pages/building-your-application/configuring/environment-variables#default-environment-variables

I have mixed feelings about it:

- On one end, this feels extra complicated, do you really want to give each of your developers the license key for their env? It's a pain to set up, it's a pain to change the key after each renewal.
- On the other end, it has the potential to allow keeping track of the use, you can count each time a developer asks for it.

I suspect that >90% of the time when developers use the env variable, they commit it to git, pointless. So maybe we should warn that we don't recommend it, unless for use cases where the key is not in git:

1. Source available, need to keep the license key private to their org.
2. The desire to have a tighter control on who in the organization can access the license key

File: docs/data/introduction/licensing/licensing.md [hunk 2]
```diff
@@ -219,6 +219,15 @@ export default function MyApp(props) {
...
+
+The license validation happens on the client side, so you must expose the environment variable to the browser.
+To do that, you can prefix it with `NEXT_PUBLIC_` as explained in [Next.js documentation](https://nextjs.org/docs/pages/building-your-application/configuring/environment-variables#bundling-environment-variables-for-the-browser).
```
[the remark this answers is not in the document]
Comment (oliviertassinari) [reply docs/data/introduction/licensing/licensing.md:225]: From how I understand it, there isn't any alternative:

```suggestion
To do that, you must prefix it with `NEXT_PUBLIC_` as explained in [Next.js documentation](https://nextjs.org/docs/pages/building-your-application/configuring/environment-variables#bundling-environment-variables-for-the-browser).
```

File: docs/data/introduction/licensing/licensing.md (hunk 1 above)
Comment (joserodolfofreitas) [reply docs/data/introduction/licensing/licensing.md:222]: Alright, I didn't intend to recommend using environment variables.
Maybe we can change the section title:

```suggestion
#### If you need to install the key as an environment variable in Next.js
```

**Comment by oliviertassinari** [comment]
@joserodolfofreitas I have pushed a commit with feels like is clearer from the perspective of a developer that knowns nothing about this.

**Comment by joserodolfofreitas** [comment]
Thanks for the contribution, @oliviertassinari!

**Pull request merged**